python/pyspark/sql/dataframe.py

```diff
@@ -1423,8 +1434,9 @@ def all_of_(xs):
             subset = [subset]
 
         # Verify we were not passed in mixed type generics."
-        if not any(all_of_type(rep_dict.keys()) and all_of_type(rep_dict.values())
-                   for all_of_type in [all_of_bool, all_of_str, all_of_numeric]):
+        if not any(key_all_of_type(rep_dict.keys()) and value_all_of_type(rep_dict.values())
+                   for (key_all_of_type, value_all_of_type)
+                   in [all_of_bool, all_of_str, all_of_numeric]):
```

Comment:
Why not just put `None` here and keep the various `all_of_*` variables defined as they were before?